src/Mentat.UI/Controllers/CardController.cs

```diff
@@ -36,83 +35,40 @@ public ActionResult Details(string id)
             return View(new CardViewModel(card));
         }
 
-        // GET: CardController/Create
-        public ActionResult Create()
+        public ActionResult AddEditCard(string id = null)
         {
-            return View();
-        }
-
-        // POST: CardController/Create
-        [HttpPost]
-        [ValidateAntiForgeryToken]
-        public ActionResult Create(IFormCollection collection)
-        {
-            try
-            {                
-                var id = Guid.NewGuid().ToString();
-                cardService.CreateCard(new Repository.Models.Card
-                {
-                  Id = id,
-                  Notes = collection["notes"],
-                  Subject = collection["subject"],
-                  Question = collection["question"],
-                  Answer = collection["answer"],
-                  DifficultyLevel = collection["difficulty_level"]
-
-                });
-                
-                return RedirectToAction(nameof(Index));
-            }
-            catch (Exception)
+            if (id == null)
             {
-                return View();
+                return View(new CardViewModel());
             }
-        }
 
-        // GET: CardController/Edit/5
-        public ActionResult Edit(string id)
-        {
-            var card = cardService.GetCard(id);
+            var card = _cardService.GetCard(id);
             if (card == null)
             {
                 return NotFound($"Card with ID = {id} not found");
             }
-
             return View(new CardViewModel(card));
         }
 
-        // POST: CardController/Edit/5
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public ActionResult Edit(string id, IFormCollection collection)
+        public ActionResult SaveCard(string id, IFormCollection collection)
```

Comment:
To fix the comments brought up in the CardService change this line to
```csharp
public IActionResult SaveCard(string id, [FromBody] Card card)
```

This will automatically serialize the parts of the form body into a Card object. You can then pass the card directly to your Card Service methods.

Reply:
This wasn't the exact change that I used, because it caused an error. But I did use something similar.
Committed similar changes. Resolving.